### ISSUE TITLE
`http_proxy` not working?

### ISSUE THREAD

**Issue by polyrabbit** [bradleyjkemp/grpc-tools#29]
Hi, could you please provide a working example?

I'm playing the great tool using the official [grpc example](https://github.com/grpc/grpc-go/tree/master/examples/helloworld). But it seems as if the client will always ignore the `http_proxy` settings.

I start `grpc-dump` in one terminal:
```bash
grpc-dump --port=12345
```

Then start the server and client, hoping the request will go through proxy `localhost:12345`:
```bash
go run greeter_server/main.go # one terminal
http_proxy=http://localhost:12345 go run greeter_client/main.go   # another terminal
```

But nothing happens on the `grpc-dump` side. And the client side works on whatever `http_proxy` settings I provide. It seems as if the `http_proxy` settings is totally ignored.

Can you help? thanks!

**Comment by bradleyjkemp** [comment]
Hi, sorry I've been meaning to add a warning about this.

Most examples (including this helloworld one) are set up as follows:
* The server listens on `localhost:1234`
* The client dials `localhost:1234`

Unfortunately proxy settings are ignored by almost all libraries if the destination is `localhost`, but you can get around this using a service like http://readme.localtest.me/ which looks like a proper domain (and so will cause the proxy to be used) but actually resolves to 127.0.0.1.

For this example, you can change the `address` variable in `greeter_client/main.go` from `localhost:50051` to `localtest.me:50051` and this should now work.

**Comment by polyrabbit** [comment]
Ah, I get it. Thanks very much!

Indeed, a warning on `localhost` connection can save a lot of confusions for newbies like me.
